core/db/default/spree/countries.rb

```diff
@@ -1,13 +1,53 @@
 require 'carmen'
 
-ActiveRecord::Base.transaction do
-  Carmen::Country.all.each do |country|
-    Spree::Country.where(iso: country.alpha_2_code).first_or_create!(
-      name: country.name,
-      iso3: country.alpha_3_code,
-      iso_name: country.name.upcase,
-      numcode: country.numeric_code,
-      states_required: country.subregions?
-    )
+# Insert Countries into the spree_countries table, checking to ensure that no
+# duplicates are created, using as few SQL statements as possible (2)
+
+connection = ApplicationRecord.connection
+
+country_mapper = ->(country) do
+  name            = connection.quote country.name
+  iso3            = connection.quote country.alpha_3_code
+  iso             = connection.quote country.alpha_2_code
+  iso_name        = connection.quote country.name.upcase
+  numcode         = connection.quote country.numeric_code
+  states_required = connection.quote country.subregions?
+
+  [name, iso3, iso, iso_name, numcode, states_required].join(", ")
+end
+
+country_values = -> do
+  carmen_countries = Carmen::Country.all
+
+  # find entires already in the database (so that we may ignore them)
+  exiting_country_isos =
+    Spree::Country.where(iso: carmen_countries.map(&:alpha_2_code)).map(&:iso)
+
+  # create VALUES statements for each country _not_ already in the database
+  carmen_countries
+  .reject { |c| exiting_country_isos.include?(c.alpha_2_code) }
+  .map(&country_mapper)
+  .join("), (")
+end
+
+country_columns = -> do
+  columns = %w(name iso3 iso iso_name numcode states_required)
+
+  if connection.adapter_name =~ /MySQL/i
+    columns.join(", ")
+  else
+    "\"#{columns.join('", "')}\""
   end
 end
+
+country_cols = country_columns.call
```

Comment:
none. just thought it would read nicely next to the following line.
will call in place